pages/blog/graphql-authz.mdx

```diff
@@ -0,0 +1,174 @@
+
+If directives are not applicable for some reason, plain JSON auth schema could be used to describe which rules should be executed for certain Object, Interface, Field, Query, or Mutation
+
+```typescript
+// defining auth schema
+
+const authSchema = {
+  Post: { __authz: { rules: ['CanReadPost'] } },
+  User: {
+    email: { __authz: { rules: ['IsAdmin'] } },
+  },
+  Mutation: {
+    publishPost: { __authz: { rules: ['CanPublishPost'] } },
+  },
+  Query: {
+    users: { __authz: { rules: ['IsAuthenticated'] } },
+  },
+  // wildcards are supported. read more: https://github.com/AstrumU/graphql-authz#wildcard-rules
+  '*': {
+    __authz: { rules: ['Reject'] },
+    '*': {
+      __authz: { rules: ['IsAuthenticated'] },
+    },
+  },
+};
+```
+
+## Comparison with GraphQL Shield
+
+[GraphQL Shield](https://github.com/maticzav/graphql-shield) is a great tool for creating authorization layers. Actually, GraphQL AuthZ was inspired by GraphQL Shield! However, it incorporates a different approach compared to GraphQL AuthZ in terms of hooking up authorization logic to the execution process.
+
+The main difference is GraphQL Shield uses field middlewares to execute authorization logic while GraphQL AuthZ wraps the entire execution process. The benefits of the wrapping approach are described in previous paragraphs however there are some drawbacks. For example, with GraphQL AuthZ post-execution rules there is no ability to fail the request earlier because post-execution rules are executed after all resolvers are executed. GraphQL Shield, on the other hand, can fail the request during the execution phase which is earlier than the post-execution phase but later than the pre-execution phase. However post-execution rules have the benefit of accessing results of all field resolvers by design. It means that the post-execution rule attached to a certain field has an access to sibling fields even if these fields are resolved by field resolvers. Also, the post-execution rule attached to a certain field which is resolved by the field resolver has an access to the field value. The middleware approach is missing these possibilities because authorization logic is executed in the context of field resolvers and has access only to the parent object which is the result of the parent field resolver or query. Results of field resolvers of the current and sibling fields are not available to authorization logic at this moment.
+
+Another great thing about GraphQL Shield is a great built-in caching mechanism. GraphQL AuthZ has no built-in caching at the moment but you could implement it by yourself outside of GraphQL AuthZ or inside of it (pull requests are more than welcome!)
+
+## Conclusion
+
+Please check out the [GraphQL AuthZ GitHub repo](https://github.com/AstrumU/graphql-authz). You can find a tutorial on how to set it up in different ways and with different technologies. Also, you can find the following working examples:
```

Comment:
```suggestion
GraphQL AuthZ is a new approach to doing GraphQL native authorization. We are happy that we can finally share this library with the community and keen to learn about the ways it might be used within your next project! Don't hesitate to contact us for questions or to evaluate whether this library might be a fit for your architecture.

Please check out the [GraphQL AuthZ repository on GitHub](https://github.com/AstrumU/graphql-authz) for learning more. You can find a tutorial on how to set it up in different ways and with different technologies.

The repository also contains the following, ready to run, examples:
```